fix(openIDConnect scheme): Check token expiration before id_token

tobyreid opened the pull request:
(Creating this pull request as it appears that it's stalled awaiting a PR into a another fork, rather than the dev branch)

Incorporates both changes from dygeenen and shollingsworth for this PR https://github.com/nuxt-community/auth-module/pull/1417

(thanks both)

This ordering is required, because it is the `access_token` is sent in the Authorization header when making an axios request.  If the `access_token` is expired, we need to refresh that token before continuing.

Checking whether the `id_token` has expired doesn't guarantee a successful call to the (a) back end and typically the `id_token` has a different (shorter) expiry schedule to the `access_token`.

dygeenen committed:
fix(openIDConnect scheme): Check token expiration before id_token

During an Axios call, the token is not automatically refreshed because it relies on the tokenExpired return of the scheme check function, which first checks id_token which often expires at the same time, and therefore does not detect any problems.
By checking first the token expiration before the id_token, the problem is solved

Related issue : 
https://github.com/nuxt-community/auth-module/issues/1370

Commit:
Fix linting error

tobyreid commented:
@Intevel is this something I can bring your attention to?

Intevel commented:
This looks clear to me. Good job! 💚

OfficialBAMM commented:
Can confirm this fixes the issue.

Using version: "5.0.0-1643365392.dae6516" with Keycloak

m-brgs commented:
@Intevel 

Hi
Can we get this merged et released please ?

Intevel commented:
I am not a maintainer, I am not authorized to merge.

m-brgs commented:
@bmulholland Hi ! Are you a maintainer or do you know someone who's able to merge this and trigger a release to npm ?

Intevel commented:
He is able to merge this, but he does all this voluntarily. A ping is just annoying

bmulholland commented:
Yep, agreed, thanks @Intevel 

On the change itself: it needs a comment explaining why the ordering needs to be as it is. That way we'll know not to accidentally break it in future. Can you please put (an abbreviated version of) the PR description as a comment in the code?

Commit:
Added comment explaining the order / reasoning

https://github.com/nuxt-community/auth-module/pull/1684#issuecomment-1070537221

Commit:
Adjusted comment to be clearer about the exact order

Commit:
One more comment edit

shollingsworth commented:
The PR has been updated with the requested comment.

Commit:
Fixing linting error (again)

Commit:
Fix linting error Check failure on line 122 in src/schemes/openIDConnect.ts

bmulholland commented:
Thanks for the fix, explanation, and for dealing with the annoying linter :)
